src/restore.ts

```diff
@@ -35,22 +35,46 @@ async function run(): Promise<void> {
             restoreKeys
         );
 
+        //Check if user wants to save cache despite of failure in any previous job
+        const saveCache = core.getInput(Inputs.SaveOnAnyFailure).toLowerCase();
+        if (saveCache == "true") {
+            core.debug(
+                `Exporting environment variable ${Variables.SaveCacheOnAnyFailure}`
+            );
+            core.exportVariable(Variables.SaveCacheOnAnyFailure, saveCache);
+            core.info(
+                `Input Variable ${Variables.SaveCacheOnAnyFailure} is set to true, the cache will be saved despite of any failure in the build.`
+            );
+        }
+
         if (!cacheKey) {
+            if (core.getInput(Inputs.FailOnCacheMiss).toLowerCase() == "true") {
```

Comment:
As mentioned in earlier comment, this can be achieved by running a separate step which fails based on the `cache-hit` output.
It would be better to start with simpler set of inputs and ask users to use outputs and conditionals to achieve their custom control. If we still see multiple asks, we can take this separately.

Reply:
Makes sense.
This feature was asked by ~10 users (issues count + thumbs up count) https://github.com/actions/cache/issues/210, https://github.com/actions/cache/issues/955

So should we drop this from our release? As correctly mentioned above, it can be easily achieved by adding an extra step in the workflow. I'm good with dropping it.

Reply:
The only issue I see is, some users are not happy with workarounds, they expect the extra workflow steps to be handled by our action, for [example](https://github.com/actions/cache/issues/852#issuecomment-1323305728), one of the user didn't like the idea of using `shell` for creating folders.

Reply:
If there are multiple asks after we ship, we would definitely think about doing it. Right now keeping simpler is easy as well.

Reply:
> We will still let those users achieve this by using the two actions in a certain way. So we are still solving their problem.

True, there is a usable workaround for it. The main appeal IMO would be that the action could provide a much better error message and reduce the boilerplate code required to get up and running. Especially if a workflow with 10+ jobs requires that the check is added in each one, it's just easy too miss or add a typo.

That all for a feature which would be fairly easy to implement.